src/lib/Type.hs

```diff
@@ -322,6 +366,12 @@ instance CoreVariant Expr where
     Case e alts _ -> do
       checkVariant e
       forM_ alts $ \(Abs _ body) -> checkVariant body
+    -- TODO: is there a way to express "this will go away once types are
+    -- monomorphic"?
```

Comment:
Well... what's a polymorphic type in our case? You mean that there are no functions with `Type` arguments left?

Reply:
Also, wouldn't `goneBy Simp` be an more accurate description?

Reply:
Polymorphic in the sense of "has arguments whose types depend on a function argument of type `Fields` (i.e. `LabeledRowKind`)". I don't think `goneBy Simp` is correct, because if the user defines a function with an argument of type `Fields`, these operations still last past `Simp` inside the `Lam` atom.

Reply:
Ugh I see... This is annoying 😕